theseus/src/util/export.rs

```diff
@@ -0,0 +1,298 @@
+#[tracing::instrument]
+#[theseus_macros::debug_pin]
+pub async fn export_mrpack(
+    profile: &Profile,
+    export_location: &Path,
+    version_id: String,
+    included_overrides: Vec<String>, // which folders to include in the overrides
+    loading_bar: bool,
+    _semaphore: &SemaphorePermit<'_>,
+) -> crate::Result<()> {
+    let profile_base_path = &profile.path;
+
+    let mut file = File::create(export_location).await?;
+    let mut writer = ZipFileWriter::new(&mut file);
+
+    // Create mrpack json configuration file
+    let packfile = create_mrpack_json(profile, version_id)?;
+    let modrinth_path_list = packfile
+        .files
+        .iter()
+        .map(|f| {
+            let path = PathBuf::from(f.path.clone());
+            let name = path.to_string_lossy();
+            let name = name.replace('\\', "/");
+            name.trim_start_matches('/').to_string()
+        })
+        .collect::<Vec<String>>();
+
+    // Build vec of all files in the folder
+    let mut path_list = Vec::new();
+    build_folder(profile_base_path, &mut path_list).await?;
+
+    // Initialize loading bar
+    let loading_bar = if loading_bar {
+        Some(
+            init_loading(
+                LoadingBarType::ZipExtract {
+                    profile_path: profile.path.to_path_buf(),
+                    profile_name: profile.metadata.name.clone(),
+                },
+                path_list.len() as f64,
+                "Exporting profile to .mrpack",
+            )
+            .await?,
+        )
+    } else {
+        None
+    };
+
+    // Iterate over every file in the folder
+    // Every file that is NOT in the config file is added to the zip, in overrides
+    for path in path_list {
+        if let Some(ref loading_bar) = loading_bar {
+            emit_loading(loading_bar, 1.0, None).await?;
+        }
+
+        // Get local path of file, relative to profile folder
+        let name = path.strip_prefix(profile_base_path)?;
+
+        // Get highest level folder pair ('a/b' in 'a/b/c', 'a' in 'a')
+        // We only go one layer deep for the sake of not having a huge list of overrides
+        let topmost_two = name
+            .iter()
+            .take(2)
+            .map(|os| os.to_string_lossy().to_string())
+            .collect::<Vec<_>>();
+
+        // a,b => a/b
+        // a => a
+        let topmost = match topmost_two.len() {
+            2 => topmost_two.join("/"),
+            1 => topmost_two[0].clone(),
+            _ => {
+                return Err(crate::ErrorKind::OtherError(
+                    "No topmost folder found".to_string(),
+                )
+                .into())
+            }
+        };
+
+        if !included_overrides.contains(&topmost) {
+            continue;
+        }
+
+        let name: std::borrow::Cow<str> = name.to_string_lossy();
+        let name = name.replace('\\', "/");
+        let name = name.trim_start_matches('/').to_string();
```

Comment:
Wouldn't it be better to do a starts with check of the path instead of trying to get the first 2 components?

Reply:
I think there are several potential cases where that could cause trouble. if we had a file without an extension like `Makefile `and a folder called `Makefile_helpers` (or anything else where there can be ambiguity between files) the rule wanting to just get Makefile would get all subfolders+files in Makefile_helpers

We could add handling for that easily dont get me wrong but it wouldnt be any simpler than using the path splitting IMO, what do you think